components/automate-ui/src/app/entities/userperms/userperms.effects.ts

```diff
@@ -65,18 +64,15 @@ export class UserPermEffects {
     withLatestFrom(this.store.select(allPerms)),
     switchMap(
       ([[action, lastTime], list]: [[GetSomeUserPerms, Date], IndexedEntities<UserPermEntity>]) => {
-        let obs: Observable<UserPermsActions>;
         const isFresh = !!list && !this.stale(lastTime);
-        if (isFresh) {
+        return isFresh ?
           // TODO: This works, but it has extraneous 'id' property in the payload.
           // Should strip that from each entry in the list.
-          obs = of(new GetSomeUserPermsSuccess(list));
-        } else {
-          this.requests.fetchSome(action.payload).pipe(
```

Comment:
The error was that the `else` block was not returning anything, so it returned "undefined".